Justfile

```diff
@@ -308,3 +308,22 @@ docs-check-generated-docs-are-current: generate-docs
       echo "Generated docs directory contains files/directories not in the repository."
       exit 1
     fi
+
+
+docs-update-tutorial-databuilder-version:
+    #!/usr/bin/env bash
+    set -euo pipefail
+
+    if [ -z ${TOKEN+x}]
+    then
+      echo "TOKEN env variable is required (a PAT with the packages:read permission)"
+      exit 1
+    fi
```

Comment:
It's a really annoying fact of the Docker registry API that you need to get a token before accessing public resources. But you can get a token even when unauthenticated, and I think it's worth doing it this way to avoid having yet another PAT to maintain.

The code looks like:
```sh
package="opensafely-core/databuilder"
token="$(
  curl --silent \
    "https://ghcr.io/token?scope=repository:$package:pull&service=ghcr.io" \
  | jq -r '.token'
)"
latest_version="$(
  curl --silent --header "Authorization: Bearer $token" \ 
    "https://ghcr.io/v2/$package/tags/list" \
  | jq -r '.tags | map(select(test("^v\\d+$"))) | max_by(. | ltrimstr("v") | tonumber)'
)"
```

Reply:
The `TOKEN` it uses in CI is the usual `GITHUB_TOKEN`; is it sensible to continue to use it there, and just do this for the if the env var isn't there, so you don't have to supply one locally?

Reply:
Actually, it looks like you can use GITHUB_TOKEN with Docker registry now
https://github.com/orgs/community/discussions/26279#discussioncomment-3251172

Reply:
I _think_ you still have to do the authentication dance because you have to exchange the GITHUB_TOKEN for a Bearer token. I'm not sure just using the GITHUB_TOKEN as is will work. But I'd be delighted to be wrong about that.

Reply:
It seems to be fine:
https://github.com/opensafely-core/databuilder/actions/runs/4343258472/jobs/7585083367

(now I'll get rid of that temporary commit!)

Reply:
Oh nice! How did you know to base64 encode it?

Reply:
It was in that link I posted :arrow_up:

Reply:
That'll teach me! I didn't actually read the link because I assumed it was the same Github discussions link I had seen on this topic earlier, but it wasn't :smile: